### ISSUE TITLE
Code does not converge

### ISSUE THREAD

**Issue by lilong-epfl** [Ling-Bao/mscnn#1]
Dear Ling,

Thank your very much for providing this open source. But I encountered some problems to run your code.
1.
![image](https://user-images.githubusercontent.com/13543618/40044563-4869fcc8-585a-11e8-8cf6-88b59b9b649f.png)
which happens when you delete the forth dimension by using the function `image_out = tf.squeeze(con_out, 3)` in the file mscnn.py
2. 
![image](https://user-images.githubusercontent.com/13543618/40044660-a68e10fa-585a-11e8-8afe-263cb984528c.png)
tmp1 will be None value in the second iteration since it cannot be assigned to other value by the while loop. This part code is not available.
3. 
![image](https://user-images.githubusercontent.com/13543618/40044732-dea2bf54-585a-11e8-8072-15e79321e7a4.png)
After I make some correction to your code, finally I can run the training code. But after hours of training, there is no sign the training goes to converge.

Is that possible to provide your pre-trained model to reproduce the result in the paper?

Best wishes,
Long LI

**Comment by Ling-Bao** [comment]
Thanks for your correction, I have corrected some problems. However, this project also have some fault that I don't have time to modify.

Not converge problem suggestion:
set large learning rate, such as 1e-1

Notes
Since I not resize datasets' image to same size, so just set batch_size param is 1.

**Comment by Ling-Bao** [comment]
The shape of mscnn model output is [batch_size, w, h, c] and c = 1. So you can use tf.squeeze(predict, 3) to solve this.

 ```
predict =  tf.squeeze(predict, 3) 
l2_loss = tf.reduce_sum((predict - label) * (predict - label))
```

**Comment by jituoo** [comment]
IThank your very much for providing this open source. But I have a problem, please help me
![1](https://user-images.githubusercontent.com/35187421/41282194-11d4cfe0-6e66-11e8-8d29-34a0efc62cc3.png)

**Comment by sratandeep16** [comment]
When running this model with my own data i came across the same error:

Incompatible shapes: [1,272,480] vs. [1,1088,1920]
	 [[{{node sub}} = Sub[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"](Squeeze, _arg_Placeholder_1_0_1/_111)]]

I have done this:
predict =  tf.squeeze(predict, 3) 
l2_loss = tf.reduce_sum((predict - label) * (predict - label))
But still it gives the same error

**Comment by Blesszd** [comment]
@sratandeep16 hi，i  miss the same problem, do you resolve?
